src/testplan/testplanJSONBasedTestLoader.ts

```diff
@@ -0,0 +1,95 @@
+import * as tmp from 'tmp';
+import * as fs from 'fs';
+import { TestSuiteInfo, TestInfo } from 'vscode-test-adapter-api';
+import { ITestPlanTestLoader } from './testplanTestLoader';
+import { ILogger } from '../logging/logger';
+
+interface BasicInfo {
+    name: string;
+    description?: string;
+    id: string;
+}
+interface Location {
+    object_name: string;
+    file: string;
+    line_no: number;
+}
+
+interface TestCaseMetadata extends BasicInfo {
+    location?: Location;
+}
+
+interface TestSuiteMetadata extends BasicInfo {
+    location?: Location;
+    test_cases: TestCaseMetadata[];
+}
+
+interface TestMetadata extends BasicInfo {
+    test_suites: TestSuiteMetadata[];
+}
+
+interface TestPlanMetadata extends BasicInfo {
+    tests: TestMetadata[];
+}
+
+export class TestPlanJSONBasedTestLoader implements ITestPlanTestLoader {
+    private tmpFile: string;
+
+    constructor(private readonly logger: ILogger) {
+        this.tmpFile = tmp.tmpNameSync({
```

Comment:
I wonder if this can be rewritten to an async code. Maybe create a factory for `ITestPlanTestLoader`s with async signature and pass a resulting file to the constructor? Similar concerns for `fs.readFileSync` in `parseOutput`, but that looks like it can just be made async.